dashboard/app/models/user.rb

```diff
@@ -1692,6 +1698,15 @@ def get_active_experiment_names
     Experiment.get_all_enabled(user: self).pluck(:name)
   end
 
+  # Returns an array of experiment name strings that a student's teachers are enrolled in
+  def get_active_experiment_names_by_teachers
+    experiments = []
+    teachers.each do |teacher|
+      experiments.concat(Experiment.get_all_enabled(user: teacher).pluck(:name))
+    end
+    experiments.uniq
+  end
```

Comment:
This is essentially the same logic we use for pilots - any of the student's teachers are enrolled in a pilot, then the student is effectively enrolled in the pilot. However that logic lives on the [script](https://github.com/code-dot-org/code-dot-org/blob/95407c68d63449d71c3958d0745be0cdea6bc826/dashboard/app/models/script.rb#L1970) and in our case we're not piloting a specific new curricular offering rather we're enabling features for students working on a existing script if their teacher is in the experiment and enables access for them.

Reply:
assuming you are planning to show special UI to students whose teachers are in `ai-tutor-toggle` experiment, this puts us in an interesting situation where the logic for passing backend experiments down to the client will not show it as enabled for students, only for teachers, because `Experiment.get_all_enabled` will only return this experiment for the teacher.

Can you please remind me which curriculum units you are planning to enable `ai-tutor-toggle` for at first? If it is just one unit, you could use a TeacherBasedExperiment instead, and be sure to specify a script_id (soon to become a requirement for performance reasons).

Reply:
to qualify a bit... I think what you have here could be fine, just want to make sure you are aware of the implications.

 If you go the route you have here in this PR, you would be using a SingleUserExperiment to add each teacher to, correct? I think this is fine, you will just have to keep in mind that the ai-tutor-toggle experiment will be magically set for teachers but not for students. as we discussed, you can still pass down the bit for whether a student has a teacher in the experiment the way you would normally pass any other server data to js.

Reply:
Could not having students officially enrolled in the experiment (and having to get that data through their teacher) impact pulling usage data/metrics?

Reply:
@Erin007 while working on this other PR:
* https://github.com/code-dot-org/code-dot-org/pull/54843
 
I realized that TeacherBasedExperiment is probably not what you want, since it doesn't let you list specific teachers, rather it is more of a percent-based teacher experiment. feel free to peek there to see the issue we ran into with SingleSectionExperiment and production DB usage issues.

For this PR, I think SingleUserExperiment (which the code here implies you are planning on using) will work fine. you will just have to keep in mind that the mechanisms described in #54843 will automatically set this experiment for you in js for teachers but not for students.

Reply:
> Could not having students officially enrolled in the experiment (and having to get that data through their teacher) impact pulling usage data/metrics? Where could the client not showing the experiment as enabled for students have negative repercussions for us?

I could be mistaken, but I'm skeptical that Amplitude would be able to figure this out using either approach to experiments. One idea would be to explicitly include this info in the Amplitude event when you log it.

Reply:
I see what you mean @ebeastlake, in [AnalyticsReporter.js](https://github.com/code-dot-org/code-dot-org/blob/663fc1e89b9e9d6cf51fa06cbde261fefd9d7bd9/apps/src/lib/util/AnalyticsReporter.js#L42) we set which experiments are enabled for the user in userProperties. You're right a student who has ai tutor access via their teacher's participation in the experiment would not show as having the experiment enabled. We'll have to pass that information to Amplitude explicitly; which I'm planning to do with we add event tracking for these components. 

I can't think of anywhere else that would be impacted by the client not showing the experiment as enabled for students since we're already showing/hiding the ai tutor panel for the student based on a property passed to the level from the server.